remote_configuration/change_applier_test.go

```diff
@@ -0,0 +1,284 @@
+package remote_configuration_test
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	cbcontainersv1 "github.com/vmware/cbcontainers-operator/api/v1"
+	"github.com/vmware/cbcontainers-operator/cbcontainers/models"
+	"github.com/vmware/cbcontainers-operator/remote_configuration"
+	"math/rand"
+	"strconv"
+	"testing"
+)
+
+func TestFeatureTogglesAreAppliedCorrectly(t *testing.T) {
```

Comment:
I think this test tried to save code by wrapping the tests with `appliedChangeTest`, iterating them and run them, but since the actual test is only:
```go
remote_configuration.ApplyConfigChangeToCR(testCase.change, &testCase.initialCR)
testCase.assertFinalCR(t, testCase.initialCR)
```

I think this wrapping and "code saving" didn't save a lot of code, but did made the test more messy and harder to read and understand. 

I think simple 4 t.Run tests with the 4 test cases will be clearer and probably will resolve with close code lines